Framework/Kernel/test/DateAndTimeTest.h

```diff
@@ -58,6 +57,10 @@ class DateAndTimeTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(d, expected);
   }
 
+  void test_constructor_fails_invalid_string() {
```

Comment:
Could we add some tests with strings that look a bit like a date time, i.e the old strings that would have passed but will now fail?